File: frontend/types/packages.ts
export type PackageQueueItem = {
  package_id: string;
  vendor_id: string;
  vendor_name: string;
  submission_date: string;
  package_status: string;
  recommendation_status: string | null;
  final_decision: string | null;
  highest_severity: string | null;
  finding_count: number;
  ocr_required_document_count: number;
  primary_finding_type: string | null;
  primary_finding_title: string | null;
  missing_required_documents: string[];
  assigned_reviewer: string | null;
  latest_reviewer_comment: string | null;
  latest_decision_at: string | null;
};

export type PackageQueueResponse = {
  items: PackageQueueItem[];
};

export type DocumentRequirementSummary = {
  required_doc_type: string;
  is_required: boolean;
  is_satisfied: boolean;
  missing_reason: string | null;
};

export type DocumentSummary = {
  document_id: string;
  doc_type: string;
  file_name: string;
  file_path: string | null;
  mime_type: string | null;
  upload_status: string;
  ocr_status: string;
  parse_status: string;
  page_count: number | null;
};

export type ExtractedFieldSummary = {
  field_id: string;
  field_name: string;
  raw_value: string | null;
  normalized_value: string | null;
  confidence: number | null;
  source_document_id: string;
  source_page: number | null;
};

export type FieldReviewOverrideSummary = {
  override_id: string;
  field_id: string;
  reviewer: string;
  original_value: string | null;
  corrected_value: string;
  correction_reason: string;
  created_at: string;
};

export type FindingSummary = {
  finding_id: string;
  finding_type: string;
  severity: string;
  title: string;
  description: string;
  suggested_action: string | null;
  requires_human_review: boolean;
  finding_status: string;
};

export type FindingResolutionSummary = {
  resolution_id: string;
  finding_id: string;
  reviewer: string;
  resolution_status: string;
  resolution_note: string;
  created_at: string;
};

export type EvidenceSummary = {
  evidence_id: string;
  evidence_type: string;
  document_id: string;
  document_name: string;
  page_num: number | null;
  snippet_text: string | null;
  related_finding_id: string | null;
};

export type ReviewDecisionSummary = {
  decision_id: string;
  reviewer: string;
  system_recommendation: string | null;
  final_decision: string;
  override_reason: string | null;
  reviewer_comment: string | null;
  decision_at: string;
};

export type ProcessingRunSummary = {
  processing_run_id: string;
  run_type: string;
  run_status: string;
  document_id: string | null;
  document_name: string | null;
  model_name: string | null;
  prompt_version: string | null;
  started_at: string;
  completed_at: string | null;
  error_message: string | null;
};

export type PackageDetailResponse = {
  package_id: string;
  vendor_id: string;
  vendor_name: string;
  package_status: string;
  submitted_at: string;
  assigned_reviewer: string | null;
  system_recommendation: string | null;
  final_decision: string | null;
  priority_score: number | null;
  notes: string | null;
  requirements: DocumentRequirementSummary[];
  documents: DocumentSummary[];
  extracted_fields: ExtractedFieldSummary[];
  field_overrides: FieldReviewOverrideSummary[];
  finding_resolutions: FindingResolutionSummary[];
  findings: FindingSummary[];
  evidence: EvidenceSummary[];
  review_history: ReviewDecisionSummary[];
  processing_runs: ProcessingRunSummary[];
};

export type ReviewDecisionCreateRequest = {
  reviewer: string;
  final_decision: string;
  reviewer_comment: string | null;
  override_reason: string | null;
};

export type ReviewDecisionCreateResponse = {
  decision_id: string;
  package_id: string;
  final_decision: string;
  package_status: string;
};

export type FieldReviewOverrideCreateRequest = {
  field_id: string;
  reviewer: string;
  corrected_value: string;
  correction_reason: string;
};

export type FieldReviewOverrideCreateResponse = {
  override_id: string;
  package_id: string;
  field_id: string;
};

export type FindingResolutionCreateRequest = {
  finding_id: string;
  reviewer: string;
  resolution_status: "resolved" | "accepted_risk" | "not_applicable";
  resolution_note: string;
};

export type FindingResolutionCreateResponse = {
  resolution_id: string;
  package_id: string;
  finding_id: string;
  finding_status: string;
};

export type SubmittedDocumentInput = {
  doc_type: string;
  file_name: string;
};

export type PackageCreateRequest = {
  vendor_name: string;
  tax_id: string | null;
  country: string | null;
  category: string | null;
  assigned_reviewer: string | null;
  submitted_documents: SubmittedDocumentInput[];
};

export type PackageCreateResponse = {
  package_id: string;
  vendor_id: string;
  package_status: string;
};

export type DocumentUploadResponse = {
  document_id: string;
  package_id: string;
  doc_type: string;
  file_name: string;
  file_path: string;
  page_count: number | null;
  parse_status: string;
};

export type PackageSimulationResponse = {
  package_id: string;
  package_status: string;
  system_recommendation: string | null;
  finding_count: number;
};

File: backend/app/schemas/packages.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class PackageQueueItem(BaseModel):
    package_id: str
    vendor_id: str
    vendor_name: str
    submission_date: datetime
    package_status: str
    recommendation_status: str | None
    final_decision: str | None
    highest_severity: str | None
    finding_count: int
    ocr_required_document_count: int
    primary_finding_type: str | None
    primary_finding_title: str | None
    missing_required_documents: list[str]
    assigned_reviewer: str | None
    latest_reviewer_comment: str | None
    latest_decision_at: datetime | None


class PackageQueueResponse(BaseModel):
    items: list[PackageQueueItem]


class DocumentRequirementSummary(BaseModel):
    required_doc_type: str
    is_required: bool
    is_satisfied: bool
    missing_reason: str | None


class DocumentSummary(BaseModel):
    document_id: str
    doc_type: str
    file_name: str
    file_path: str | None
    mime_type: str | None
    upload_status: str
    ocr_status: str
    parse_status: str
    page_count: int | None


class ExtractedFieldSummary(BaseModel):
    field_id: str = ""
    field_name: str
    raw_value: str | None
    normalized_value: str | None
    confidence: float | None
    source_document_id: str
    source_page: int | None


class FindingSummary(BaseModel):
    finding_id: str
    finding_type: str
    severity: str
    title: str
    description: str
    suggested_action: str | None
    requires_human_review: bool
    finding_status: str = "open"


class EvidenceSummary(BaseModel):
    evidence_id: str
    evidence_type: str
    document_id: str
    document_name: str
    page_num: int | None
    snippet_text: str | None
    related_finding_id: str | None


class ReviewDecisionSummary(BaseModel):
    decision_id: str
    reviewer: str
    system_recommendation: str | None
    final_decision: str
    override_reason: str | None
    reviewer_comment: str | None
    decision_at: datetime


class ProcessingRunSummary(BaseModel):
    processing_run_id: str
    run_type: str
    run_status: str
    document_id: str | None
    document_name: str | None
    model_name: str | None
    prompt_version: str | None
    started_at: datetime
    completed_at: datetime | None
    error_message: str | None


class FieldReviewOverrideSummary(BaseModel):
    override_id: str
    field_id: str
    reviewer: str
    original_value: str | None
    corrected_value: str
    correction_reason: str
    created_at: datetime


class FindingResolutionSummary(BaseModel):
    resolution_id: str
    finding_id: str
    reviewer: str
    resolution_status: str
    resolution_note: str
    created_at: datetime


class PackageDetailResponse(BaseModel):
    package_id: str
    vendor_id: str
    vendor_name: str
    package_status: str
    submitted_at: datetime
    assigned_reviewer: str | None
    system_recommendation: str | None
    final_decision: str | None
    priority_score: int | None
    notes: str | None
    requirements: list[DocumentRequirementSummary]
    documents: list[DocumentSummary]
    extracted_fields: list[ExtractedFieldSummary]
    field_overrides: list[FieldReviewOverrideSummary] = Field(default_factory=list)
    finding_resolutions: list[FindingResolutionSummary] = Field(default_factory=list)
    findings: list[FindingSummary]
    evidence: list[EvidenceSummary]
    review_history: list[ReviewDecisionSummary]
    processing_runs: list[ProcessingRunSummary]


class ReviewDecisionCreateRequest(BaseModel):
    reviewer: str = Field(min_length=1, max_length=120)
    final_decision: str = Field(min_length=1, max_length=60)
    reviewer_comment: str | None = Field(default=None, max_length=2000)
    override_reason: str | None = Field(default=None, max_length=1000)


class ReviewDecisionCreateResponse(BaseModel):
    decision_id: str
    package_id: str
    final_decision: str
    package_status: str


class FieldReviewOverrideCreateRequest(BaseModel):
    field_id: str = Field(min_length=1, max_length=120)
    reviewer: str = Field(min_length=1, max_length=120)
    corrected_value: str = Field(min_length=1, max_length=2000)
    correction_reason: str = Field(min_length=3, max_length=2000)


class FieldReviewOverrideCreateResponse(BaseModel):
    override_id: str
    package_id: str
    field_id: str


class FindingResolutionCreateRequest(BaseModel):
    finding_id: str = Field(min_length=1, max_length=120)
    reviewer: str = Field(min_length=1, max_length=120)
    resolution_status: Literal["resolved", "accepted_risk", "not_applicable"]
    resolution_note: str = Field(min_length=3, max_length=2000)


class FindingResolutionCreateResponse(BaseModel):
    resolution_id: str
    package_id: str
    finding_id: str
    finding_status: str


class SubmittedDocumentInput(BaseModel):
    doc_type: str = Field(min_length=1, max_length=80)
    file_name: str = Field(min_length=1, max_length=255)


class PackageCreateRequest(BaseModel):
    vendor_name: str = Field(min_length=1, max_length=255)
    tax_id: str | None = Field(default=None, max_length=60)
    country: str | None = Field(default="US", max_length=60)
    category: str | None = Field(default=None, max_length=120)
    assigned_reviewer: str | None = Field(default=None, max_length=120)
    submitted_documents: list[SubmittedDocumentInput] = Field(default_factory=list)


class PackageCreateResponse(BaseModel):
    package_id: str
    vendor_id: str
    package_status: str


class DocumentUploadResponse(BaseModel):
    document_id: str
    package_id: str
    doc_type: str
    file_name: str
    file_path: str
    page_count: int | None
    parse_status: str


class PackageSimulationResponse(BaseModel):
    package_id: str
    package_status: str
    system_recommendation: str | None
    finding_count: int
